test/Performance/run-perftests.py

```diff
@@ -0,0 +1,238 @@
+    result_xml_path = os.path.join(SCRIPT_ROOT_PATH, '{}.xml'.format(runid))
+
+    saved_path = os.environ.get('PATH')
+    print("# Prepending {dir} to PATH".format(dir = os.path.dirname(cli_path)))
+    os.environ['PATH'] = os.path.dirname(cli_path) + ';' + os.environ.get('PATH')
+    try:
+        run_command(
+            dotnet_path, 'run', '-p', runner_src_path, '-c', 'Release', '--',
+            '-runner', cli_path, '-runid', runid,
+            '-runnerargs', 'test {json} -c Release'.format(json = PERFTEST_JSON_PATH),
+            title = "Run {id}".format(id = runid),
+            from_dir = SCRIPT_ROOT_PATH,
+        )
+        if not os.path.exists(result_xml_path):
+            raise FatalError("Running {id} seems to have failed: {xml} was not generated".format(
+                id = runid, xml = result_xml_path
+            ))
+    finally:
+        print("# Reverting PATH")
+        os.environ['PATH'] = saved_path
+
+def compare_results(base_id, test_id, out_html, xunitperf_src_path):
+    analyzer_path = get_xunitperf_analyzer_path(xunitperf_src_path)
+
+    # Make sure there aren't any stale XMLs in the target dir
+    for xml in glob.glob(os.path.join(SCRIPT_ROOT_PATH, '*.xml')):
+        if not os.path.basename(xml) in [base_id + '.xml', test_id + '.xml']:
+            os.rename(xml, xml + '.bak')
+
+    try:
+        run_command(
+            analyzer_path, SCRIPT_ROOT_PATH, '-compare', base_id, test_id, '-html', out_html,
+            title = "Generate comparison report",
+            from_dir = SCRIPT_ROOT_PATH,
+        )
+        if os.path.exists(out_html):
+            print("# Comparison finished, please see \"{report}\" for details.".format(report = out_html))
+        else:
+            raise FatalError("Failed to genererate comparison report: \"{report}\" not found.".format(report = out_html))
+
+    finally:
+        # Revert the renamed XMLs
+        for xml in glob.glob(os.path.join(SCRIPT_ROOT_PATH, '*.xml.bak')):
+            os.rename(xml, xml[0:-4])
+
+def main():
+    try:
+        process_arguments()
+        check_requirements()
+
+        script_args.xunit_perf_path = os.path.realpath(script_args.xunit_perf_path)
+
+        clone_repo(XUNITPERF_REPO_URL, script_args.xunit_perf_path)
+        make_xunit_perf(script_args.xunit_perf_path)
+
+        base_runid = script_args.runid + '.base'
+        test_runid = script_args.runid + '.test'
+        out_html = os.path.join(SCRIPT_ROOT_PATH, script_args.runid + '.html')
+
+        run_perf_test(test_runid, script_args.test_cli, script_args.xunit_perf_path)
+        if script_args.base_cli != None:
+            run_perf_test(base_runid, script_args.base_cli, script_args.xunit_perf_path)
+            compare_results(base_runid, test_runid, out_html, script_args.xunit_perf_path)
+
+    except FatalError as error:
+        print("! ERROR: {msg}".format(msg = error.message))
+        return 1
+
+if __name__ == "__main__":
+    sys.exit(main())
```

Comment:
Tried it out, and it does seem to produce a 0 exit code, may be good to be explicit though 😄